### PR TITLE
reportbutton: Add Game Ticks option

### PR DESCRIPTION
**Pull request by mlvzk**
Counts game ticks since login and displays them on the Report button

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/reportbutton/ReportButtonPlugin.java [hunk 1]
```diff
@@ -61,6 +62,8 @@
 	private static final DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter.ofLocalizedTime(FormatStyle.MEDIUM);
 	private static final DateFormat DATE_FORMAT = new SimpleDateFormat("MMM. dd, yyyy");
 
+	private int tickCounter;
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/reportbutton/ReportButtonPlugin.java]: Please put this inbetween ``loginTime`` and ``ready`` and without newlines like so:
```java
private Instant loginTime;
private int tickCounter;
private boolean ready;
```

File: runelite-client/src/main/java/net/runelite/client/plugins/reportbutton/ReportButtonPlugin.java [hunk 2]
```diff
@@ -113,13 +116,21 @@ public void onGameStateChanged(GameStateChanged event)
 			case LOGGED_IN:
 				if (ready)
 				{
+					tickCounter = 0;
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/reportbutton/ReportButtonPlugin.java]: Place this after ``loginTime = Instant.now();``

File: runelite-client/src/main/java/net/runelite/client/plugins/reportbutton/ReportButtonPlugin.java (hunk 1 above)
Comment (devLotto) [reply runelite-client/src/main/java/net/runelite/client/plugins/reportbutton/ReportButtonPlugin.java]: Maybe it should also be renamed to something along the lines of ``ticksSinceLogin`` to be more clear about what it does.

File: runelite-client/src/main/java/net/runelite/client/plugins/reportbutton/TimeStyle.java [hunk 3]
```diff
@@ -31,7 +31,8 @@
 	LOGIN_TIME("Login Timer"),
 	UTC("UTC Time"),
 	JAGEX("Jagex HQ Time"),
-	LOCAL_TIME("Local Time");
+	LOCAL_TIME("Local Time"),
+	TICK("Game Ticks");
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/reportbutton/TimeStyle.java]: Change ``TICK`` to ``GAME_TICKS``

**Comment by mlvzk** [comment]
Thank you, I commited the changes. I'm not sure if `updateReportButtonTime()` should be in `onGameTick`, should I at least add an `if` clause that runs it only if `config.time()` is `GAME_TICKS`?

**Comment by devLotto** [comment]
Yeah, ``updateReportButtonTime`` should probably be in onGameTick like you currently have it, so that the tick count is updated when the tick changes. It won't have the same updating behaviour as the other time styles, but those don't require the same accuracy either.

You should probably add the ``config.time() == GAME_TICKS`` check to onGameTick.

**Comment by mlvzk** [comment]
Changed, it's wrapped with `if` now

**Pull request merged**